google/cloud/internal/rest_stub_helpers.cc

```diff
@@ -30,8 +30,13 @@ Status RestResponseToProto(google::protobuf::Message& destination,
   auto json_to_proto_status =
       google::protobuf::util::JsonStringToMessage(*json_response, &destination);
   if (!json_to_proto_status.ok()) {
-    return Status{
-        StatusCode::kInternal, std::string{json_to_proto_status.message()}, {}};
+    ErrorInfo err_info("Failure creating proto Message from Json",
+                       "google-cloud-cpp",
+                       {{"message_type", destination.GetTypeName()},
+                        {"json_string", *json_response}});
+    return Status{StatusCode::kInternal,
```

Comment:
Should we try to preserve `json_to_proto_status.code()`?  [It looks like `absl::StatusCode` is already compatible with `google::cloud::StatusCode` via their mutual compatibility with the gRPC codes.]

Reply:
I think I like @devbww idea of preserving the status code sounds sensible, maybe consider using `google::cloud::internal::ErrorInfoBuilder`